contracts/liquidity_hub/incentive-manager/src/helpers.rs

```diff
@@ -0,0 +1,167 @@
+use std::cmp::Ordering;
+
+use cosmwasm_std::{
+    ensure, BankMsg, Coin, CosmosMsg, Decimal, MessageInfo, OverflowError, OverflowOperation,
+    Uint128,
+};
+
+use white_whale_std::incentive_manager::{Config, IncentiveParams, DEFAULT_INCENTIVE_DURATION};
+
+use crate::ContractError;
+
+/// Processes the incentive creation fee and returns the appropriate messages to be sent
+pub(crate) fn process_incentive_creation_fee(
+    config: &Config,
+    info: &MessageInfo,
+    incentive_creation_fee: &Coin,
+    params: &mut IncentiveParams,
+) -> Result<Vec<CosmosMsg>, ContractError> {
+    let mut messages: Vec<CosmosMsg> = vec![];
+
+    // verify the fee to create an incentive is being paid
+    let paid_fee_amount = info
+        .funds
+        .iter()
+        .find(|coin| coin.denom == incentive_creation_fee.denom)
+        .ok_or(ContractError::IncentiveFeeMissing)?
+        .amount;
```

Comment:
It can't be done that way because must_pay checks that you send only one coin. That would be the case if you are creating an incentive with the same asset you pay the incentive creation with, otherwise you will send two assets which must_pay will give an error with.

Reply:
Great point!